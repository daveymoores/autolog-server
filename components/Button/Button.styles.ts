import css from "styled-jsx/css";

export default css`
  button {
    height: 44px;
    padding: 0 3em;
    border: none;
    font-weight: 500;
  }

  @media (prefers-color-scheme: dark) {
    button {
      background-color: var(--lightGreen);
      color: var(--darkGrey);
    }

    .button--invert {
      background-color: var(--darkGrey);
      color: var(--lightGreen);
      border: var(--lineWidth) solid var(--lightGreen);
    }
  }

  @media (prefers-color-scheme: light) {
    button {
      background-color: var(--darkGrey);
      color: var(--lightGreen);
    }

    .button--invert {
      background-color: var(--lightGreen);
      color: var(--darkGrey);
      border: var(--lineWidth) solid var(--darkGrey);
    }
  }
`;
