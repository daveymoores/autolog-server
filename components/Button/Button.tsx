import React, { ButtonHTMLAttributes } from "react";

import styles from "../Button/Button.styles";

interface ButtonProps extends Partial<ButtonHTMLAttributes<any>> {
  text: string;
}

const Button: React.FC<ButtonProps> = ({ type, text, onClick }) => (
  <>
    <button type={type} onClick={onClick}>
      {text}
    </button>
    <style jsx>{styles}</style>
  </>
);

export default Button;
