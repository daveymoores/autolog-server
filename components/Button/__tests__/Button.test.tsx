import React from "react";
import renderer from "react-test-renderer";

import Button from "../Button";

describe("Button", () => {
  it("renders button unchanged", () => {
    const tree = renderer.create(<Button text={"Click here"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
