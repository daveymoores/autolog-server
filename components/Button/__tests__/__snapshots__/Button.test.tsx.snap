// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button unchanged 1`] = `
<button
  className="jsx-836649993"
/>
`;
